### PR TITLE
Support array<key, value>

### PR DESCRIPTION
**Pull request by brendt**

### CODE REVIEW

**Comment by brendt** [comment]
@mfn would you like to review?

File: src/DocblockFieldValidator.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 class DocblockFieldValidator extends FieldValidator
 {
-    public const DOCBLOCK_REGEX = '/@var ((?:(?:[\w?|\\\\<>])+(?:\[])?)+)/';
+    public const DOCBLOCK_REGEX = '/@var ((?:(?:[\w?|\\\\<>,\s])+(?:\[])?)+)/';
```
Comment (mfn) [inline src/DocblockFieldValidator.php:9]: Personally this exceeds my capability to quickly understand the regex, no matter how good I'm with them 😄

File: tests/DataTransferObjectTest.php [hunk 2]
```diff
@@ -421,11 +421,38 @@ public function iterable_is_supported()
...
+        };
+    }
+
+    /** @test */
+    public function key_value_arrays_key_is_type_checked()
+    {
+        $this->expectException(DataTransferObjectError::class);
+
+        new class(['map' => [1 => 1, 'b' => 2]]) extends DataTransferObject {
+            /** @var array<string, int> */
```
Comment (mfn) [inline tests/DataTransferObjectTest.php:446]: Maybe also add a test showing the `array<string,int>` (i.e _no_ space) also works. Just in case.

**Review by mfn — APPROVED** [review]
LGTM!

Smaller suggestions provided mostly for the tests

**Comment by mfn** [comment]
PS: no idea what's with the github action phpstan error in the PR, i.e. https://github.com/spatie/data-transfer-object/actions/runs/225264569 => will take a look later

File: src/DocblockFieldValidator.php (hunk 1 above)
Comment (brendt) [reply src/DocblockFieldValidator.php:27]: I did my best to add some docs :D

**Comment by brendt** [comment]
Released as 2.4.0: https://github.com/spatie/data-transfer-object/releases/tag/2.4.0

**Pull request merged**